topostats/tracing/dnatracing.py

```diff
@@ -723,11 +787,16 @@ def trace_image(
     # Check both arrays are the same shape
     assert image.shape == grains_mask.shape
 
-    cropped_images, cropped_masks = prep_arrays(image, grains_mask, pad_width)
+    cropped_images, cropped_masks, grain_crop_coordinates = prep_arrays(image, grains_mask, pad_width)
     n_grains = len(cropped_images)
     LOGGER.info(f"[{filename}] : Calculating statistics for {n_grains} grains.")
     n_grain = 0
     results = {}
+    all_global_traces = {}
+    all_trace_heights = {}
+    all_trace_cumulative_distances = {}
+    trace_plot_data = {}
+    trace_plot_data["cropped_grains"] = {}
     for cropped_image, cropped_mask in zip(cropped_images, cropped_masks):
         result = (
```

Comment:
Don't need these parentheses, it puts the dictionary within a tuple and means you then have to index `result[0]` to get at the actual results.